### ISSUE TITLE
Remove the PeerStore from the API of identify and kademlia

### ISSUE THREAD

**Issue by tomaka** [libp2p/rust-libp2p#216]
Instead of using a `Peerstore` trait, use a trait or an API more specific to identify or kademlia.

**Comment by tomaka** [comment]
And relay.

**Comment by tomaka** [comment]
No longer relevant after #723
